File: Data Structures & Algorithms/duplicate-integer/submission-3.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        // we can solve this with a hashSet
        // we add all the items into a set
        HashSet<Integer> numberSet = new HashSet<>();

        // go through the list of numbers
        for (int i = 0; i < nums.length; i++) {
            // check if the current number is in the hashSet
            if (numberSet.contains(nums[i])) {
                return true;
            }
            // add the number into the set
            numberSet.add(nums[i]);
        }

        return false;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones/submission-3.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        curr = 0 
        ans = 0 

        for number in nums:
            if number == 1:
                curr += 1 
                ans = max(curr, ans)
            
            else:
                curr = 0
        
        return ans